src/dhcp_mon.cpp

```diff
@@ -78,6 +98,132 @@ static void signal_callback(evutil_socket_t fd, short event, void *arg)
+        event_active(ev_tx_cache_counter_update, 0, 0);
+    }
+}
+
+/**
+ * @code update_cache_counter(evutil_socket_t fd, short event, void *arg);
+ *
+ * @brief Callback function to update cache counter from DB counter.
+ *
+ * @param fd        libevent socket
+ * @param event     event triggered
+ * @param arg       pointer to user provided context (libevent base)
+ *
+ * @return none
+ */
+static void update_cache_counter(evutil_socket_t fd, short event, void *arg) {
+    if (write_counter_to_db != 0b00) {
+        syslog(LOG_ERR, "Write DB counter from cache counter hasn't stop, cannot sync cache counter from DB counter\n");
+        return;
+    }
+    dhcp_packet_direction_t* dir_ptr = reinterpret_cast<dhcp_packet_direction_t*>(arg);
+    dhcp_packet_direction_t dir = *dir_ptr;
+    std::string dir_str = (dir == DHCP_RX ? "RX" : "TX");
+
+    syslog(LOG_INFO, "Update %s cache counter\n", dir_str.c_str());
+
+    auto match_pattern = COUNTERS_DB_COUNTER_TABLE_PREFIX + downstream_if_name + "*";
+    std::lock_guard<std::mutex> lock(write_counter_mutex);
+    auto keys = mCountersDbPtr->keys(match_pattern);
+
+    // Store interfaces in DB counter table
+    std::unordered_set<std::string> updated_intfs;
+
+    auto counter_map = dhcp_device_get_counter(dir);
+    for (auto &itr : keys) {
+        std::string interface;
+        std::string vlan;
+        auto first = itr.find_first_of(":");
+        auto last = itr.find_last_of(":");
+        if (first == last) {
+            // Vlan interfaces
+            interface = itr.substr(first + 1, itr.length() - first);
+            vlan = itr.substr(first + 1, itr.length() - first);
+        } else {
+            // Physical interfaces
+            interface = itr.substr(last + 1, itr.length() - last);
+            vlan = itr.substr(first + 1, last - first - 1);
+        }
+
+        if (vlan.compare(downstream_if_name) != 0) {
+            syslog(LOG_INFO, "Skip [%s - %s] since it's not related to current downstream interface\n", vlan.c_str(), interface.c_str());
+            continue;
+        }
+
+        auto interface_key = construct_counter_db_table_key(interface);
+        auto counter_json = mCountersDbPtr->hget(interface_key, dir_str);
+        std::replace(counter_json.get()->begin(), counter_json.get()->end(),'\'', '\"');
+        Json::Value root;
+        if (!parse_json_str(counter_json.get(), &root)) {
+            syslog(LOG_WARNING, "Failed to parse DB counter for %s, skip sync it from DB counter to cache counter", interface.c_str());
+            continue;
+        }
+
+        auto counter = counter_map->find(interface);
+        if (counter == counter_map->end()) {
+            // Didn't have this interface in counter map, re-init it
+            syslog(LOG_INFO, "Didn't find %s in cache counter, init it now\n", interface.c_str());
+            initialize_cache_counter(*counter_map, interface);
+        }
+
+        for (int i=0; i < DHCP_MESSAGE_TYPE_COUNT; i++) {
```

Comment:
`int i=0` 

The format is not aligned with other place. Do you have any tool to have you re-format the code?

Reply:
Addressed